src/Providers/FeatureLayer.js

```diff
@@ -95,7 +95,11 @@ export var FeatureLayerProvider = FeatureLayerService.extend({
       queryString.push(field + " LIKE upper('%" + text + "%')");
     }
 
-    return queryString.join(' OR ');
+    if (this.options.where) {
+      return this.options.where + ' AND ' + queryString.join(' OR ');
```

Comment:
i'm happy to make the suggested change, but can you clarify exactly what is necessary to generate inappropriate behavior?  

everything is working fine on my end.